pkg/auth/dependency/oauth/handler/result_authz.go

```diff
@@ -84,3 +93,19 @@ func (a authorizationResultRequireAuthn) WriteResponse(rw http.ResponseWriter, r
 func (a authorizationResultRequireAuthn) IsInternalError() bool {
 	return false
 }
+
+func redirect(rw http.ResponseWriter, redirectURI string) {
+	rw.WriteHeader(http.StatusFound)
+	rw.Header().Set("Location", redirectURI)
```

Comment:
WriteHeader should be called AFTER setting all headers.

```
rw.Header().Set("Location", redirectURI)
rw.Header().Set("Content-Type", "text/html; charset=utf-8")
rw.WriteHeader(http.StatusFound)
// Write body.
```

Reply:
yes, thanks for reminding 🤦🏻‍♀️